### ISSUE TITLE
[SF_ACC][Table]: Focused row of a table embeded in ObjectPage is not visible when using up arrow key to move focus up

### ISSUE THREAD

**Issue by i323808** [SAP/ui5-webcomponents-react#7570]
### Bug Description

Render a `Table` in the `ObjectPage` component and press the up arrow key to move the focus up, the focused row is invisible when the scrollbar is moved up.

### Expected Behaviour

The focused row is supposed to be visible when scroll up likes using the down arrow key to scroll down.

### Isolated Example

https://stackblitz.com/edit/github-jjmatnyo-vvdgdtuo?file=src%2FApp.tsx

### Steps to Reproduce

1. navigate to https://stackblitz.com/edit/github-jjmatnyo-vvdgdtuo?file=src%2FApp.tsx 
2. scroll down to the bottom of ObjectPage
3. focus a table row
4. press up arrow key to move the focus up
5. the focused row is invisible when the scrollbar is moved up

### Priority

None

### UI5 Web Components Version

2.12.0

### Browser

Chrome

### Declaration

- [x] I’m not disclosing any internal or sensitive information.

**Comment by NHristov-sap** [comment]
Hello @SAP/ui5-webcomponents-table  ,

Please take a look on this issue.

Best Regards,
Nikolay Hristov

**Comment by aborjinik** [comment]
Hi colleagues,
Removing the [onBlur handler](https://github.com/SAP/ui5-webcomponents-react/blob/45eeb1d2f02dcf8962d5b35b4cf9822da489404d/packages/main/src/components/ObjectPage/index.tsx#L749-L754) from the ObjectPage so that it sets the scrollPaddingBlock but does not remove it resolves the issue.
Could you please take a look?

**Comment by Lukas742** [comment]
Hi @aborjinik 

The `ObjectPage` needs to programmatically set the scroll-padding via JS, because otherwise, if items in the header area are focused, the scroll behavior is incorrect.

When investigating the issue, I noticed that when navigating the table via arrow keys, the `relatedTarget` is always `null`. Since we check for the `relatedTarget` to reset the scroll padding, this causes the described behavior.

[ObjectPage/index.tsx](https://github.com/SAP/ui5-webcomponents-react/blob/5664dd5c5c9df44722dbbf1ca84de174467826b8/packages/main/src/components/ObjectPage/index.tsx#L749C9-L754)
```
          if (opNode && !e.currentTarget.contains(e.relatedTarget as Node)) { // as the `relatedTarget` is always null, this check is falsely `true`
            opNode.style.scrollPaddingBlock = '0px';
          }
```

I compared the behavior to the `ui5-list` and there the `relatedTarget` is available. Could you please check whether this is intended?

Here you can see the difference of the List and Table component: https://stackblitz.com/edit/github-jjmatnyo-3ad7jzd1?file=src%2FApp.tsx

**Comment by aborjinik** [comment]
Hi Lukas,

The Table component follows a different architectural approach compared to other web components. Specifically, the accessibility role is applied directly to the custom element itself, rather than to an internal element within the shadow DOM as it is common with other components. This design enables full use of ARIA attributes, such as setting IDRef relationships, which would otherwise be restricted due to shadow DOM encapsulation. In fact, this architectural shift was the foundation for how the new Web Components Table initiative began: https://github.com/SAP/ui5-webcomponents/issues/4702

Since the role is assigned to the custom element, it must be the element that receives focus but if you put `tabindex="-1"` on a custom element with shadow DOM enabled, it makes all of the child elements non-tabbable. (This is actually unexpected, since tabindex usually does not propagate, and makes it so only people using a mouse can click the elements.) To address this, table rows intentionally do not use `tabindex="-1"` (unlike ListItems), meaning they are not focusable by default during rendering, but can become focusable when needed. As a result, there is no `relatedTarget` during focus transitions. But I don't consider this a table-specific issue. You would encounter the same behavior if, for example, you added a Text component and attempted to click on it, or any other component that follows a similar approach.

Hope this clarifies the table behavior. What is the reason for removing scroll-padding-top?

**Comment by Lukas742** [comment]
Thanks for the explanation Cahit!

> What is the reason for removing scroll-padding-top?

We need to programmatically remove the scroll padding because, otherwise, the content scrolls when elements in the header or footer area are focused. This behavior is due to the current structure of the `ObjectPage` component, which was originally created back when IE11 support was necessary. Although the structure has been improved over time with new APIs, a full refactor would require changing the base structure. Since that would be a significant effort, we don’t have the capacity to achieve this in the foreseeable future.


Here you can see the difference:

Without resetting `scrollPaddingBlock`:

https://github.com/user-attachments/assets/83327182-5cb6-476c-90ea-fd1f03416f76

With resetting `scrollPaddingBlock`:

https://github.com/user-attachments/assets/960518e6-501f-4005-9f8c-dbbbf17a408d

___

Do you know if the `relatedTarget` being unavailable is a limitation only for the `Table` and its subcomponents, or do other UI5 web components have the same behavior? I’m asking because I’m looking for a workaround, and if only the table is affected, it would make things much easier 😄

**Comment by Lukas742** [comment]
@aborjinik quick update: 
I've implemented a fallback in the linked PR.
I had to rely on `requestAnimationFrame` to get the reference of the next focused element after `blur`, but as the callback is executed right before the next repaint, the focus __should__ always be updated already.
In case you have a better idea, or see a way how the `Table` could be extended to support `relatedTarget`, please let me know. 🙂
